server/tserver/src/main/java/org/apache/accumulo/tserver/tablet/Tablet.java

```diff
@@ -2730,6 +2730,37 @@ public void updateTabletDataFile(long maxCommittedTime, FileRef newDatafile, Fil
 
   }
 
+  public void updateLastLocation(){
+    synchronized (timeLock) {
+      Tablet t = tabletServer.getOnlineTablet(extent);
+      SortedMap<FileRef, DataFileValue> sm = t.getDatafiles();
+      for (Map.Entry<FileRef, DataFileValue> m : sm.entrySet()) {
+        FileRef fileref = m.getKey();
+        DataFileValue dfv = m.getValue();
+        MasterMetadataUtil.updateLastLocation(getTabletServer().getContext(), extent, fileref,
+                dfv, tabletTime.getMetadataTime(persistedTime),
+                tabletServer.getClientAddressString(), tabletServer.getLock(),
+                lastLocation);
+      }
+      lastLocation = null;
+    }
+  }
+
+  public boolean needsLastUpdate() {
+    boolean needsUpdate = false;
+    if (isClosing() || isClosed()) {
+      return false;
+    }
+    Tablet t = tabletServer.getOnlineTablet(extent);
+    //will keep looking for a better way to check if a tablet is old and needs to be updated or not. In the mean time, this seems to work and the property can be adjusted.
+    if (tabletTime.getAndUpdateTime() - t.tabletTime.getMetadataTime(persistedTime).getTime() >= tabletServer.getConfiguration().getCount(Property.TSERV_LASTLOCATION_UPDATE_TIME)) {
+      //updat persisted time, maybe not needed
+      persistedTime = tabletTime.getAndUpdateTime();
```

Comment:
I would think we would do this when we do the update, not when we are checking whether we are to update.

Reply:
Only did it here since I pass in the updated persistedTime to updateLastLocation. I suppose it could be moved and simplify the check function.